bin/propolis-server/src/lib/server.rs

```diff
@@ -458,6 +461,15 @@ async fn instance_ensure(
         }));
     }
 
+    let log = server_context.log.clone();
+    let services = Arc::clone(&server_context.services);
+    tokio::task::spawn(async move {
+        // Once the VmController has signaled that it is shutting down,
+        // we'll clean up the per-instance service providers as well.
+        let _ = stop_recv.await;
```

Comment:
So, this could return an `Err` meaning the Sender was dropped without sending. That could happen if the `vm_state_worker` thread panics. I don't think we ever actually mark the instance as `Failed` but here seems like maybe one place?

Reply:
Was going to open an issue, but there's already https://github.com/oxidecomputer/propolis/issues/209